lookout/style/format/analyzer.py

```diff
@@ -293,6 +243,122 @@ def train(cls, ptr: ReferencePointer, config: Mapping[str, Any], data_service: D
+                # triggered rule description
+            )
+
+            file_comments.append(
+                generate_comment(
+                    filename=file.path,
+                    line=line_number,
+                    text=text,
+                    confidence=self._get_comment_confidence(
+                        line_ys, line_ys_pred, line_winners, rules)))
+
+        return file_comments
+
+    @staticmethod
+    def _get_comment_confidence(line_ys: Sequence[int], line_ys_pred: Sequence[int],
+                                line_winners: Sequence[int], rules: Rules) -> int:
+        confidence = 0
+        confidence_count = 0
+        for yi, y_predi, winner in zip(line_ys, line_ys_pred, line_winners):
+            if yi != y_predi:
+                confidence += rules.rules[winner].stats.conf
+                confidence_count += 1
+        return int(round(confidence * 100 / confidence_count))
+
+    @staticmethod
+    def _group_line_nodes(y: Sequence[int], y_pred: Sequence[int], vnodes_y: Sequence[VirtualNode],
+                          new_vnodes: Sequence[VirtualNode], rule_winners: Sequence[int]
+                          ) -> Tuple[int, Tuple]:
+        """
+        Group virtual nodes and related lists from feature extractor by line number.
+
+        It yields line number and sublists of corresponding items from all input sequences.
+        Line sublists are skipped in case there is no difference in predicted and original labels.
+        Line sublists are merged in case new line on the end was replaced by target without
+        newline. It is a helper function for `FormatAnalyser._generate_file_comments()`
+
+        :param y: Sequence of original labels.
+        :param y_pred: Sequence of predicted labels by the model.
+        :param vnodes_y: Sequence of the labeled `VirtualNode`-s corresponding to labeled samples.
+        :param new_vnodes: Sequence of all the `VirtualNode`-s corresponding to the input with \
+                           applied predictions. `CodeGenerator.apply_predicted_y()` is used for \
+                           that.
+        :param rule_winners: List of rule winners.
+        :return: 1-based line number and sublists of corresponding items from all input sequences.
+        """
+        line_ys, line_ys_pred, line_vnodes_y, line_winners = [], [], [], []
+        generate_comment = False
+        vnodes_index = 0
+        for yi, y_predi, vnode_y, winner in zip(y, y_pred, vnodes_y, rule_winners):
+            if not line_ys or vnode_y.start.line == line_vnodes_y[0].start.line:
+                # collect all nodes on the same line
+                line_ys.append(yi)
+                line_ys_pred.append(y_predi)
+                line_vnodes_y.append(vnode_y)
+                line_winners.append(winner)
+                if yi != y_predi:
+                    generate_comment = True
+                continue
+            else:
```

Comment:
else + if -> elif

Reply:
there is lines under `else`, but  not in `if`: https://github.com/src-d/style-analyzer/pull/335/files/78858ed7fa0dfd36046531fe0adefe92e020d0dc#diff-b71e0e71a29c83c6fa2ee406a76c3baeR358